### PR TITLE
Unify multi remote test and moved them to functional

### PR DESCRIPTION
**Pull request by danimtb**
Changelog: omit
Docs: omit

While looking for some tests doing stuff with multiple remotes I realized we had two different files for similar tests. This PRs unifies them and move the test their corresponding *functional/remote* folder

### CODE REVIEW

File: conans/test/functional/remote/multi_remote_test.py [hunk 1]
```diff
@@ -2,13 +2,96 @@
...
+        sleep(1)  # For timestamp and updates checks
+
+        # Download Hello0 from local with client_b
+        client_b.run("install Hello0/0.0@lasote/stable -r local --build missing")
+        client_b.run("remote list_ref")
+        self.assertIn(": local", str(client_b.user_io.out))
+
+        # Update Hello0 with client_a and reupload
+        self._create(client_a, "Hello0", "0.0", modifier="\n")
+        client_a.run("upload Hello0/0.0@lasote/stable -r local")
```
Comment (jgsogo) [inline conans/test/functional/remote/multi_remote_test.py]: Check the output to know it has been effectively uploaded

File: conans/test/functional/remote/multi_remote_test.py [hunk 2]
```diff
@@ -2,13 +2,96 @@
...
+
+        # Execute info method in client_b, should advise that there is an update
+        client_b.run("info Hello0/0.0@lasote/stable -u")
+        self.assertIn("Recipe: Update available", client_b.out)
+        self.assertIn("Binary: Cache", client_b.out)
+
+        # Now try to update the package with install -u
+        client_b.run("remote list_ref")
+        self.assertIn(": local", str(client_b.user_io.out))
+        client_b.run("install Hello0/0.0@lasote/stable -u --build")
```
Comment (jgsogo) [inline conans/test/functional/remote/multi_remote_test.py]: Check that it has been updated

File: conans/test/functional/remote/multi_remote_test.py [hunk 3]
```diff
@@ -2,13 +2,96 @@
...
+        sleep(1)  # For timestamp and updates checks
+        self._create(client_a, "Hello0", "0.0", modifier="\n\n")
+        client_a.run("upload Hello0/0.0@lasote/stable -r default")
+
+        # Now client_b checks for updates without -r parameter
+        client_b.run("info Hello0/0.0@lasote/stable -u")
+        self.assertIn("    Remote: local", str(client_b.user_io.out))
+        self.assertIn("    Recipe: Cache", client_b.out)
+
+        # But if we connect to default, should tell us that there is an update IN DEFAULT!
```
Comment (jgsogo) [inline conans/test/functional/remote/multi_remote_test.py]: Check that it is telling us about the update

File: conans/test/functional/remote/multi_remote_test.py [hunk 4]
```diff
@@ -2,13 +2,96 @@
...
+        client_b.run("remote list_ref")
+        self.assertIn(": local", str(client_b.user_io.out))
+
+        # Well, now try to update the package with -r default -u
+        client_b.run("install Hello0/0.0@lasote/stable -r default -u --build")
+        self.assertIn("Hello0/0.0@lasote/stable: Calling build()",
+                      str(client_b.user_io.out))
+        client_b.run("info Hello0/0.0@lasote/stable -u")
+        self.assertIn("Recipe: Cache", client_b.out)
+        self.assertIn("Binary: Cache", client_b.out)
```
Comment (jgsogo) [inline conans/test/functional/remote/multi_remote_test.py]: And check the `remote list_ref`... has it changed? is it associated with `local` or `default`?

**Comment by danimtb** [comment]
@jgsogo Didn't want to spend so much time and mess with the changes, but I will take a look no that you reviewed them

**Comment by danimtb** [comment]
Changes were accomplished 🕴

**Pull request merged**